zipkin-cassandra/src/main/scala/com/twitter/cassie/connection/ClusterClientProvider.scala

```diff
@@ -141,7 +141,11 @@ private[cassie] class ClusterClientProvider(
         // call login
         val loginMap: java.util.Map[String, String] = HashMap("username" -> username, "password" -> password)
         val authRequest = new AuthenticationRequest(loginMap)
-        client.login(authRequest)
+        val future = client.login(authRequest)
+        future onFailure { exception =>
+          throw exception
+        }
```

Comment:
I want the exception viewable in the output (which gets turned into the log).  The thinking was that throwing the exception would make it viewable.  Otherwise what's returned is just a future and the exception is hidden.  

The for comprehension looks like a good refactor, but it doesn't seem to make it more testable because this `CassandraThriftFramedCodec` gets buried in multiple levels of futures. 

What I wanted to do initially was to use scalatest to assert the right exception is raised using intercept.  But because of the multiple levels of futures the exception doesn't get bubbled up.
```scala
intercept[SomeException] {
  ...code...
} 

So the question is: [Whats a good way to test this method?]
  * one idea is to just make a unit test and see if the exception gets thrown but that doesn't seem to mesh with the rest of the testing strategy which seems more integration-ish